### ISSUE TITLE
 New issue Compiled CSS gives incorrect paths when using django-storages and S3 with non-default paths

### ISSUE THREAD

**Issue by KevinBrolly** [torchbox/wagtail#895]
Hi,

I have an issue much like https://github.com/torchbox/wagtail/issues/134 I am using django-storages and s3 but the compiled css gives the wrong path for font files.  My S3/django-storages is slightly different however as I store static and media files in the same bucket but in different folders, as explained in this stack overflow comment: http://stackoverflow.com/a/10825691/429396

So the fix for the other issue which is to make the scss variables for static-root and css-root ../../ as it is here https://github.com/torchbox/wagtail/blob/master/wagtail/wagtailadmin/static/wagtailadmin/scss/variables.scss#L8 will not work for me as I have another folder "static" in place.

**Comment by gasman** [comment]
Hi @KevinBrolly,

I can't see how that setup would cause a problem. The CSS and fonts are both stored within the 'static' location, and the relative path never goes outside that folder - the 'media' location should never enter into it, as that's only used for user-uploaded files.

Once you've run the 'compress' and 'collectstatic' management commands, you should have a folder structure like:

    static
        CACHE
            css
                some-css-file.css
        wagtailadmin
            scss
                fonts
                    Bitter-Bold.eot

Consequently, when some-css-file.css refers to "../../wagtailadmin/scss/fonts/Bitter-Bold.eot", it should arrive at the right place regardless of whatever else is in your S3 bucket. Is that not the case for you?

(Also, have you seen #633? That would be my first suspect for font problems when using S3, as it's a not-so-obvious and not very well publicised effect of remote font loading.)

**Comment by KevinBrolly** [comment]
Okay, it was me being daft.

I was settings COMPRESS_STORAGE = 'storages.backends.s3boto.S3BotoStorage' but setting STATICFILES_STORAGE = 'utils.s3utils.StaticS3BotoStorage' as per the stackoverflow link I posted.  Meaning the CACHE folder was going in the root of the bucket while the staticfiles were going in /static/ as per my StaticS3BotoStorage class.

I set COMPRESS_STORAGE = STATICFILES_STORAGE and all is well!

Closing issue!
